### PR TITLE
added a spec to make sure no functions are kicked off when they should not be

### PR DESCRIPTION
**Pull request by nmccready**
Not sure if this is worth much. But I added this test to double check some foobars I had locally. This was to verify functionality.

### CODE REVIEW

File: test/watch.js [hunk 1]
```diff
@@ -44,6 +44,24 @@ describe('gulp', function() {
...
 
+    it('should not call the function when no file changes: no options', function(done) {
+      var tempFile = path.join(outpath, 'watch-func.txt');
+      var called = false;
+
+      createTempFile(tempFile);
+
+      var watcher = gulp.watch(tempFile, function(cb) {
+        watcher.end();
+        called = true;
```
Comment (callumacrae) [inline test/watch.js]: Throw an error or call `should.fail()` here instead, "expected true to equal false" isn't a descriptive error

**Comment by callumacrae** [comment]
I don't really like the way this test was written: 200ms seems a really long time to delay the tests, I feel a smaller amount of time like 10ms would still be more than enough.

**Comment by nmccready** [comment]
Thats fine I can make the time faster.

**Comment by nmccready** [comment]
I mean it could look better but I didn't want to add dependencies like sinon.

**Comment by callumacrae** [comment]
by the `should.fail()` comment, I meant this:

```js
    it('should not call the function when no file changes: no options', function(done) {
      var tempFile = path.join(outpath, 'watch-func.txt');

      createTempFile(tempFile);

      var watcher = gulp.watch(tempFile, function() {
        should.fail('Watcher erroneously called');
      });

      setTimeout(done, 10);
    });
```

**Comment by nmccready** [comment]
ahh thanks

File: test/watch.js [hunk 2]
```diff
@@ -44,6 +44,21 @@ describe('gulp', function() {
...
+      var tempFile = path.join(outpath, 'watch-func.txt');
+
+      createTempFile(tempFile);
+
+      var watcher = gulp.watch(tempFile, function() {
+        should.fail('Watcher erroneously called');
+      });
+
+      setTimeout(function() {
+        watcher.end();
```
Comment (nmccready) [inline test/watch.js]: still need the watcher end.

Comment (callumacrae) [inline test/watch.js]: fair enough! I guessed most of it

**Comment by phated** [comment]
We have an open PR to switch to chokidar so I don't think this is going to be merged first.  It will probably need to be updated when that PR is merged.

**Comment by nmccready** [comment]
I will happily update when needed. Let me know.

**Comment by phated** [comment]
@nmccready we've switched to chokidar, if you are able to update this, it would be much appreciated. Thanks!

**Comment by nmccready** [comment]
10-4

**Comment by nmccready** [comment]
@phated rebased and fixed changes

**Comment by callumacrae** [comment]
All looks good to me.

**Comment by nmccready** [comment]
Btw where / what project and commits was chokidar pulled into. I am seeing some tasks not clean up and exit now when everything is done. IE gulp process continues. I am looking to analyze the code to check for possible problems, or new design pattern I might be missing. 

BTW no watches are running on this task series so that is what is more confusing.

```
[12:38:00] Finished 'karmaNoCoverage' after 18 s
[12:38:00] Finished 'frontendNoCoverageSpec' after 18 s
[12:38:00] Finished 'rebuildSpec' after 49 s
//hangs here never happened prior to 7243d6ead83c2e18236b9323be4a9265dcd511b6
```

**Comment by nmccready** [comment]
More info:
```
 ├─┬ rebuildSpec
[12:45:41] │ └─┬ <series>
[12:45:41] │   ├─┬ <parallel>
[12:45:41] │   │ ├── commonSpec
[12:45:41] │   │ └── backendSpec
[12:45:41] │   ├── gulpSpec
[12:45:41] │   ├─┬ <parallel>
[12:45:41] │   │ ├─┬ otherAssets
[12:45:41] │   │ │ └─┬ <parallel>
[12:45:41] │   │ │   ├─┬ webroot
[12:45:41] │   │ │   │ └─┬ <series>
[12:45:41] │   │ │   │   └── <anonymous>
[12:45:41] │   │ │   ├─┬ vendor
[12:45:41] │   │ │   │ └─┬ <parallel>
[12:45:41] │   │ │   │   ├── vendor_scripts
[12:45:41] │   │ │   │   ├── vendor_css
[12:45:41] │   │ │   │   ├── vendor_fonts
[12:45:41] │   │ │   │   └── vendor_assets
[12:45:41] │   │ │   └── <anonymous>
[12:45:41] │   │ ├─┬ angular
[12:45:41] │   │ │ └─┬ <parallel>
[12:45:41] │   │ │   ├── styles
[12:45:41] │   │ │   ├── markup
[12:45:41] │   │ │   └── browserify
[12:45:41] │   │ └─┬ angularAdmin
[12:45:41] │   │   └─┬ <parallel>
[12:45:41] │   │     ├── stylesAdmin
[12:45:41] │   │     ├── markupAdmin
[12:45:41] │   │     └── browserifyAdmin
[12:45:41] │   └─┬ frontendNoCoverageSpec
[12:45:41] │     └─┬ <series>
[12:45:41] │       └── karmaNoCoverage
```

**Comment by nmccready** [comment]
Actually I have an easier task with fewer tasks now that hangs.

out put via grep

- gulp spec > gulpout.log
- cat gulpout.log | grep Starting
- cat gulpout.log | grep Finished

```
[13:31:21] Starting 'spec'...
[13:31:21] Starting 'parallel'...
[13:31:21] Starting 'commonSpec'...
[13:31:21] Starting 'backendSpec'...
[13:31:21] Starting 'frontendSpec'...
[13:31:21] Starting 'karma'...
[13:31:44] Starting 'gulpSpec'...
[13:31:37] Finished 'commonSpec' after 16 s
[13:31:39] Finished 'backendSpec' after 18 s
[13:31:44] Finished 'karma' after 24 s
[13:31:44] Finished 'frontendSpec' after 24 s
[13:31:44] Finished 'parallel' after 24 s
[13:31:45] Finished 'gulpSpec' after 861 ms
[13:31:45] Finished 'spec' after 24 s
//hang
```

**Comment by nmccready** [comment]
I just noticed the hang issue is happening on our master so disregard. It is probably something we're foobaring somewhere.

**Comment by phated** [comment]
@nmccready I am pretty sure it is karma, I've seen that problem before where it is keeping a connection open internally

**Comment by nmccready** [comment]
Anything else I need to do here?

**Comment by phated** [comment]
@nmccready nope, this looks fine.  Now that we switched to chokidar, it is really just testing chokidar but it doesn't hurt anything to have it in the suite.  Thanks!

**Pull request merged**